components/mysql/mysql.app.mjs

```diff
@@ -0,0 +1,331 @@
+    },
+    whereValues: {
+      type: "string[]",
+      label: "Values",
+      description: "This is the list of **values** that will match every `?` symbol in the **where expression**. If you want to build yourself the **values** (eg. `{{[\"string1\", \"string2\"]}}`)",
+    },
+    whereOperator: {
+      type: "string",
+      label: "Where Operator",
+      description: "Build a filter based on this operator (eg. `=, >, >=, <, !=, <=, LIKE`).",
+      options() {
+        return [
+          {
+            label: "Equal to",
+            value: "=",
+          },
+          {
+            label: "Greater than",
+            value: ">",
+          },
+          {
+            label: "Greater than or equal to",
+            value: ">=",
+          },
+          {
+            label: "Less than",
+            value: "<",
+          },
+          {
+            label: "Not equal to",
+            value: "!=",
+          },
+          {
+            label: "Less than or equal to",
+            value: "<=",
+          },
+          {
+            label: "Simple pattern matching",
+            value: "like",
+          },
+        ];
+      },
+    },
+    whereValue: {
+      type: "string",
+      label: "Value",
+      description: "This is the value to be set after the selected operator",
+    },
+    storedProcedure: {
+      type: "string",
+      label: "Stored Procedure",
+      description: "List of stored procedures in the current database",
+      async options() {
+        return this.listStoredProcedures();
+      },
+    },
+    storedProcedureParameters: {
+      type: "string[]",
+      label: "Parameters",
+      description: "Parameters for the stored procedure",
+      optional: true,
+    },
+  },
+  methods: {
+    async getConnection() {
+      const {
+        host,
+        port,
+        username,
+        password,
+        database,
+      } = this.$auth;
+      return mysqlClient.createConnection({
+        host,
+        port,
+        user: username,
+        password,
+        database,
+      });
+    },
+    async closeConnection(connection) {
+      const connectionClosed = new Promise((resolve) => {
+        connection.connection.stream.on("close", resolve);
+      });
+      await connection.end();
+      await connectionClosed;
+    },
+    async executeQuery({
+      connection, preparedStatement,
+    }) {
+      const [
+        result,
+      ] = await connection.execute(preparedStatement);
+      return result;
+    },
+    async executeQueryConnectionHandler(preparedStatement) {
+      let connection;
+
+      try {
+        connection = await this.getConnection();
+
+        return await this.executeQuery({
+          connection,
+          preparedStatement,
+        });
+
+      // eslint-disable-next-line no-useless-catch
+      } catch (error) {
+        throw error;
```

Comment:
There is some specific reason to have a catch just with a `throw`? In my view, you just need the `finally`